app/lib/event_store/services/service_deleted_event.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class EventStore::Services::ServiceDeletedEvent < EventStore::Event
+  default_scope -> { where(event_type: ::Services::ServiceDeletedEvent) }
+
+  scope :by_service_id, lambda { |service_id|
+    concat_query = if System::Database.oracle?
+                     "('%service_id: ' || #{service_id} || '%')"
```

Comment:
Why does it have to be concatenated in the database?

Reply:
because it is inside a subquery @guicassolato and it is either this or load them all in memory and select one by one with Ruby afterwards.

Reply:
I don't get this. The if condition implies that the scope will always reduce to either

```ruby
->(service_id) { where("data LIKE ('%service_id: ' || #{service_id} || '%')") }
```
or
```ruby
->(service_id) { where("data LIKE CONCAT('%service_id: ',  #{service_id}, '%')") }
```

Either way, the actual SQL will never be built unless you provide the scope with a value for `service_id`, meaning that the scope could be simply:

```ruby
->(service_id) { where("data LIKE ('%service_id: #{service_id},%')") }
```

Notice the comma (`,`) added right before the last `%` to prevent `service_id: 10` from being selected when in fact we wanted only `service_id: 1`.

Reply:
Right!! Good one!! 💪 🥇